### ISSUE TITLE
XDP program not getting cleaned up after loading and unloading

### ISSUE THREAD

**Issue by anfredette** [redhat-et/bpfd#146]
To start, the dispatcher and my xdp program are not in `bpftool prog list`

Execute:
```
bpfctl load /home/afredette/projects/xdp-tutorial/basic01-xdp-pass/xdp_pass_kern.o -p xdp -i vboxnet1 --priority 30 -s "xdp"
```
Followed by:
```
bpfctl unload --iface vboxnet1 4d63d526-81eb-476c-812d-335f5ac81785
```
Then, I still see the dispatcher and program in `bpftool prog list`
```
$ sudo bpftool prog list

[SNIP]

156: xdp  name dispatcher  tag 94d5f00c20184d17  gpl
	loaded_at 2022-09-15T16:57:14-0400  uid 0
	xlated 672B  jited 531B  memlock 4096B  map_ids 66
	btf_id 358
157: ext  name xdp  tag 3b185187f1855c4c  gpl
	loaded_at 2022-09-15T16:57:14-0400  uid 0
	xlated 16B  jited 19B  memlock 4096B
	btf_id 350
```
Is this expected?

**Comment by dave-tucker** [comment]
Should be fixed by #147

**Comment by astoycos** [comment]
Fixed by #147
